client/core/wallet.go

```diff
@@ -197,34 +200,51 @@ func (w *xcWallet) connected() bool {
 // flag to true, and validates the deposit address. Use Disconnect to cleanly
 // shutdown the wallet.
 func (w *xcWallet) Connect() error {
+	reloadWallet := func() {
+		wallet, err := asset.OpenWallet(w.AssetID, w.cfg, w.logger, w.net)
+		if err != nil {
+			w.logger.Errorf("error opening wallet: %v", err)
+			return
+		}
+		w.Wallet = wallet
```

Comment:
Modifying this field would seem to be racy, but consumers won't be using any other wallet methods until the wallet has been connected.  I believe it's safe to assume that if correct access patterns with the `xcWallet` methods will be sufficient.  I don't seen any issues jumping out based on the code, have not observed any races in testing.

Reply:
This seems iffy. Almost certainly OK for our UI. Dims any prospects for concurrent use though. I'm just wondering if it'd be better to replace the entire `xcWallet`.

Reply:
Thought about that too, but you run into similar concurrency issues, just higher up.  Worth a shot though.

Reply:
Hmm, maybe `(c *Core) connectWallet(w *xcWallet)` could create a new `xcWallet` if `Connect` failed, and pop it into the `c.wallets` map...

Reply:
I think that actually worked out fairly well.  It's more involved because there are many more paths to `connectWallet` than just `connectedWallet`, but in the end I think this may have simplified things somewhat.